app/frontend/actions/voting.tsx

```diff
@@ -52,23 +55,28 @@ export default (store: Store) => {
     }
     loadingAction(state, 'Preparing transaction...')
     state = getState()
-
+    const sourceAccountInfo = getSourceAccountInfo(state)
+    if (!sourceAccountInfo.stakingXpub?.xpubHex) {
```

Comment:
for these runtime errors I would rather use a standard assert(condition) statement - these shouldn't really happen at runtime and should rather be "technical" assertions and are more concise than "if ... then ... throw".

We have an `assert()` helper for which you can find in app/src/utls/assertion.ts. Same for other places where you introduced this pattern. Sorry for not claryfing that earlier